src/librustc_target/abi/mod.rs

```diff
@@ -672,6 +676,7 @@ impl FieldsShape {
 
     pub fn offset(&self, i: usize) -> Size {
         match *self {
+            FieldsShape::Primitive => unreachable!(),
```

Comment:
it's not unreachable, it's actually fairly easy to reach. Use `bug!("`FieldsShape::offset` can only be invoked on aggregates")`